pkg/network/mptpool/pool.go

```diff
@@ -0,0 +1,90 @@
+package mptpool
+
+import (
+	"sync"
+
+	"github.com/nspcc-dev/neo-go/pkg/util"
+)
+
+// Pool stores unknown MPT nodes along with the corresponding paths.
+type Pool struct {
+	lock   sync.RWMutex
+	hashes map[util.Uint256]*Item
```

Comment:
But I still would like to keep mptpool structure, because we need request retransmission mechanism. So now `storage.MemoryStore` is a part of mptpool.